### ISSUE TITLE
有几个小问题向你请教下。

### ISSUE THREAD

**Issue by DCC-lzhy** [shengode503/Segmentation-Based-Deep-Learning-Approach-for-Surface-Defect-Detection-Pytorch#1]
首先谢谢你开源了你的版本，我在GitHub上搜了这个论文的title，一共有个代码库，除了TensorFlow的官方版本，你这款是最完美的，其次是mengcius/Surface-Defect-Detection，它是在seanXYZ/SegDecNet基础上建立的，前者在train_segment中注释到`# Loss functions
criterion_segment  = torch.nn.MSELoss() # mean squared error (squared L2 norm) 
criterion_segment  = torch.nn.BCEWithLogitsLoss() # non convergence`，
一个问题是他们的seg网络最后的conv1x1没有加BN，前者的notebook记录了训练日志，mseloss确实收敛了；另一个是他们没有可视化seg出来的图，你的仓库里有。
我自己在zacario-li/Segmentation-based-deep-learning-approach-for-surface-defect-detection基础上修改的，我按TensorFlow版本的训练方式，奇偶次训练正负样本，bcewithlogitsloss确实不收敛，我还没试diceloss。
关于你的代码里，

1. 在models.model.Decision_net中，#64用了sigmoid,  #L77-L79没有用sigmoid，是不是有点问题？
2. [https://github.com/shengode503/Segmentation-Based-Deep-Learning-Approach-for-Surface-Defect-Detection-Pytorch/blob/afda7ba60038a32ad4c5590388026c415813844f/lib/data_helper/data_loader.py#L14-L16](url)这边的数据list pkl是怎么样得到的，和TensorFlow版本的一样吗？
3. pics.result.png里有两个fp，第一个fp是tp吧

**Comment by DCC-lzhy** [comment]
已解决，bcewithlogitsloss也是可以收敛的，单是验证集（TensorFlow版本的64分）瑕疵的iou能到0.56，后续训练的decision network能precision=1，recall=1.

**Comment by RangeLiu** [comment]
> 已解决，bcewithlogitsloss也是可以收敛的，单是验证集（TensorFlow版本的64分）瑕疵的iou能到0.56，后续训练的decision network能precision=1，recall=1.

请问您的bce loss是怎样做到收敛的？我这尝试过那段代码，确实不收敛。
